### DISCUSSION TITLE
Removing deployment does not work

### DISCUSSION THREAD

**Discussion by rauno56** [vercel/community#1511] [Help] [answered]
### Summary

Any time I call `vercel rm <deployment>`, I get 

```
Error: Deployment <deployment id> not found
```

1. `vercel ls <project>`, pick a Deployment url `https://<...>.vercel.app`
2. `vercel rm https://<...>.vercel.app`, returns the error above.
3. Both `vercel inspect https://<...>.vercel.app` and `vercel inspect <deployment id>` work fine and return the data.

### Additional information

Vercel CLI 28.15.3

**Comment by amyegan** [comment]
Thanks for reaching out @Rauno56. Are you using the specific deployment [generated URL](https://vercel.com/docs/concepts/deployments/generated-urls#) (i.e. `<project-name>-<unique-hash>-<scope-slug>.vercel.app`), or the always-up-to-date alias URL (i.e. `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`)?

I'm was able to reproduce the error if I try to delete a deployment via the alias. But I am able to delete a deployment in my project using the unique URL

**Comment by rauno56** [comment]
Does `vercel ls` ever list aliases?

I've never tried `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`, only the ones that have unique hash in them.

In conclusion:

- `vercel rm <project-name>-<unique-hash>-<scope-slug>.vercel.app` errors.
- `vercel rm dpl_<longer deployment id / hash>` errors.
- `vercel rm <project-name>-git-<branch-name>-<scope-slug>.vercel.app` - not sure, my CI deletes them automatically, never needed to delete one.

**Reply by johnm** [reply]
I'm running into the same problem.

@amyegan Perhaps you're thinking we're looking to solve a difference use case?
The whole point of using the unique hash version (or deployment id) is to clean up all of the old, stale deployments that are lingering in a given project's deployment list.

[And spending time nuking them all manually is silly.]

**Reply by amyegan** [reply]
Can you share some specific deployment IDs or unique hash URLs? I'm still not able to reproduce the error (`vercel rm <project-name>-<unique-hash>-<scope-slug>.vercel.app` works fine for me), but I can use a deployment id or deployment url to look up more details about what is happening in your case 🙏

**Comment by johnm** [comment, marked as answer]
I talked with Support.

The confusing (and information leaking) part is that `vercel inspect` works even when one has not `vercel switch`’d into the project name.

On top of that, the error messages for commands like `vercel rm` are very unhelpful in not suggesting things like “blah not found. Have you ’switched` to the project?”.

And the docs need to be updated to be cross-linked with contextual information like this.

Thanks,
John

**Reply by amyegan** [reply]
I just found your support case, and I see that your feedback was shared with the CLI team as well. Thanks for sharing this info and pointing out that this was missing in our documentation 🙇

**Comment by AmGhGitHub** [comment]
I still have this issue!
